### PR TITLE
Feature/multiplatform settings persistence

### PR DESCRIPTION
**Pull request by rodvar**
- Implementation of https://github.com/bisq-network/bisq-mobile/issues/39
 - key value store common implementation based on most used KMP lib
 - provided common tests with mocked implementation
 - using SettingsRepository for an example of real usage: url and connected flag gets saved now and loaded every time the app starts (xClients)
 - tested on the 3 apps, using keychain for iOS

### CODE REVIEW

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/Settings.kt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package network.bisq.mobile.domain.data.model
 
+import kotlinx.serialization.Serializable
+
+@Serializable
```
Comment (HenrikJannsen) [inline bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/Settings.kt:5]: Not form your change, but I would rename `bisqUrl` to a more clear name. Like `bisqApiUrl`

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    override suspend fun save(item: T) {
+        settings[keyPrefix + item.id] = serializer(item)
+    }
+
+    override suspend fun saveAll(items: List<T>) {
+        items.forEach { save(it) }
+    }
+
+    override suspend fun get(id: String?): T? {
```
Comment (HenrikJannsen) [inline bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt:25]: Why is id nullable?

**Review by HenrikJannsen — COMMENTED** [review]
To me it is not clear why an id with null is permitted. Then the prefix is the only map key, no?

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/Settings.kt (hunk 1 above)
Comment (rodvar) [reply bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/model/Settings.kt:5]: might be a good opportunity to do so, I'll have a look

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt (hunk 2 above)
Comment (rodvar) [reply bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt:25]: I was using it for a case where you are saving a single obj of a type but now that I think it doesn't really makes sense and could be error-prone, I'll change it soon.

**Comment by rodvar** [comment]
rebased to latest main changes

File: bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt (hunk 2 above)
Comment (rodvar) [reply bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt]: found a better solution, is pushed now

**Comment by rodvar** [comment]
rebased branch to latest main changes

**Pull request merged**